iroh-relay/src/server/client.rs

```diff
@@ -98,43 +100,30 @@ impl ClientConn {
             send_queue: send_queue_r,
             disco_send_queue: disco_send_queue_r,
             node_gone: peer_gone_r,
-            key,
-            preferred: false,
-            server_channel: server_channel.clone(),
+            node_id,
+            clients: clients.clone(),
         };
 
         // start io loop
         let io_done = done.clone();
-        let io_client_id = client_id;
+        let io_client_id = node_id;
         let handle = tokio::task::spawn(
             async move {
-                let (key, conn_num) = io_client_id;
-                let res = actor.run(io_done).await;
-
-                // remove the client when the actor terminates, no matter how it exits
-                let _ = server_channel
-                    .send(actor::Message::RemoveClient {
-                        node_id: key,
-                        conn_num,
-                    })
-                    .await;
-                match res {
+                let key = io_client_id;
+                match actor.run(io_done).await {
                     Err(e) => {
-                        warn!(
-                            "connection manager for {key:?} {conn_num}: writer closed in error {e}"
-                        );
+                        warn!("connection manager for {key:?}: writer closed in error {e:?}");
```

Comment:
- Put the `key` in the span on this task as a field: `remote_node = %key.short_fmt()` or something.
- Use alt formatting for the error: `{e:#}`.

Reply:
done